jedi/evaluate/docstrings.py

```diff
@@ -55,14 +59,49 @@ def _search_param_in_numpydocstr(docstr, param_str):
                 m = re.match('([^,]+(,[^,]+)*?)(,[ ]*optional)?$', p_type)
                 if m:
                     p_type = m.group(1)
-
-                if p_type.startswith('{'):
-                    types = set(type(x).__name__ for x in literal_eval(p_type))
-                    return list(types)
-                else:
-                    return [p_type]
+                return _expand_typestr(p_type)
         return []
 
+    def _search_return_in_numpydocstr(docstr):
+        r"""
+        Search `docstr` (in numpydoc format) for type(-s) of `param_str`.
+        """
+        doc = NumpyDocString(docstr)
+        returns = doc._parsed_data['Returns']
+        returns += doc._parsed_data['Yields']
+        found = []
+        for p_name, p_type, p_descr in returns:
+            if not p_type:
+                p_type = p_name
+                p_name = ''
+
+            m = re.match('([^,]+(,[^,]+)*?)$', p_type)
+            if m:
+                p_type = m.group(1)
+            found.extend(_expand_typestr(p_type))
+        return found
+
+
+def _expand_typestr(p_type):
+    """
+    Attempts to interpret the possible types
+    """
+    # Check if alternative types are specified
+    if re.search('\\bor\\b', p_type):
+        types = [t.strip() for t in p_type.split('or')]
+    # Check if type has a set of valid literal values
+    elif p_type.startswith('{'):
+        if not is_py3:
+            # python2 does not support literal set evals
+            # workaround this by using lists instead
+            p_type = p_type.replace('{', '[').replace('}', ']')
+        types = set(type(x).__name__ for x in literal_eval(p_type))
```

Comment:
Numpydoc allows types to be a set of literals like:
```
order : {'C', 'F', 'A'}
```
It works like this:
```
>>> from ast import literal_eval
>>> set(type(x).__name__ for x in literal_eval("['x', [], 2.3]"))
{'float', 'list', 'str'}
```

This code in already in docstrings.py:
https://github.com/davidhalter/jedi/blob/dev/jedi/evaluate/docstrings.py#L60

All this being said I'm sure that there is already a way to do this in jedi